src/sentry/api/paginator.py

```diff
@@ -354,9 +357,15 @@ def get_result(self, limit, cursor=None):
         offset = cursor.offset if cursor is not None else 0
         # Request 1 more than limit so we can tell if there is another page
         data = self.data_fn(offset=offset, limit=limit + 1)
-        has_more = (len(data) == limit + 1)
-        if has_more:
-            data.pop()
+
+        if isinstance(data, list):
+            has_more = len(data) == limit + 1
+            if has_more:
+                data.pop()
+        elif isinstance(data.get('data'), list):
+            has_more = len(data['data']) == limit + 1
+            if has_more:
+                data['data'].pop()
```

Comment:
you should probably have an
```python
else:
    raise NotImplementedError
```
or something here

Reply:
I don't think we want to have an error if there aren't any pages left to paginate through. 
Just tried it out, and it raises "an error occurred" 

The cursor updates itself based on the `has_more` value, returning a boolean in the `paginationLinks` object in responseHeaders, so I don't think we need it.

Reply:
shouldn't it still be a list/object even if it's an empty one? what i'm saying is that if you don't hit either of these conditions, won't you get an error below when `has_more` hasn't been defined? so seems better to be explicit.

Reply:
Oh I see, I'll add an else statement to make sure it's defined. Thank you!!